compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1020,6 +1021,12 @@ pub trait ToPredicate<'tcx> {
     fn to_predicate(self, tcx: TyCtxt<'tcx>) -> Predicate<'tcx>;
 }
 
+impl<'tcx> ToPredicate<'tcx> for Predicate<'tcx> {
```

Comment:
Tiny nit but haha where is this needed?

Reply:
It's needed for this line: https://github.com/rust-lang/rust/pull/101212/files#diff-5a5306a398f9ba6fae2220777e8ca62df2daf3529308a86662096577d3065214R2023